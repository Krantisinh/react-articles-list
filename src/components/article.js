/* eslint-disable react/prop-types */
import React from "react";

const articleStyle = {
  container: {
    padding: "1rem",
    display: "flex",
    flexDirection: "column",
    marginBottom: "1.5rem",
    fontFamily: "Verdana, Arial, Helvetica, sans-serif",
    border: "1px solid #ccc",
  },
  date: {
    fontSize: "0.75rem",
    marginBottom: "0.5rem",
  },
  body: {
    marginBottom: "1rem",
  },
  author: {
    fontWeight: "bold",
    textAlign: "end",
  },
};

export const Article = (props) => {
  const article = props.article;
  const author = props.author;
  return (
    <div style={articleStyle.container}>
      <h2>{article.title}</h2>
      <div style={articleStyle.date}>{article.date}</div>
      <div style={articleStyle.body}>{article.body}</div>
      <div style={articleStyle.author}>
        - {author.firstName + " " + author.lastName}
      </div>
    </div>
  );
};
