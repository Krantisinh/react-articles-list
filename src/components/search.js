import React, { memo, useState } from "react";

import PropTypes from "prop-types";

const SearchStyle = {
  textContainer: {
    margin: "1rem 0",
  },
};

const Search = (props) => {
  const [value, setValue] = useState("");

  const handleInput = (event) => {
    const val = event.target.value;
    setValue(val);
    props.onSearch(val);
  };

  return (
    <div style={SearchStyle.textContainer}>
      <input type="text" value={value} onInput={handleInput}></input>
    </div>
  );
};

Search.propTypes = {
  onSearch: PropTypes.func,
};

export const SearchM = memo(Search);
