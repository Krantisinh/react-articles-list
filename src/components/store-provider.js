import React, { useContext } from "react";

import { StoreContext } from "../state/context";

export const StoreProvider = (Component) => {
  const context = useContext(StoreContext);
  console.log(context);
  const state = context.state;

  const WithStore = (props) => <Component {...props} state={state}></Component>;
  WithStore.displayName = `${Component.name}Container`;

  return WithStore;
};
