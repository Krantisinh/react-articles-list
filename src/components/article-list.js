/* eslint-disable react/prop-types */
import React from "react";
import { Article } from "./article";

export const ArticleList = (props) => {
  return (
    <div>
      {Object.keys(props.articles).map((id) => (
        <Article
          article={props.articles[id]}
          actions={props.actions}
          key={id}
        />
      ))}
    </div>
  );
};
