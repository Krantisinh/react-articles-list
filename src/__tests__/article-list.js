import { expect, jest } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";
import { ArticleList } from "../components/article-list";

const testProps = {
  articles: {
    a: {},
    b: {},
  },
  actions: {
    getArticleAuthor: jest.fn(() => ({})),
  },
};

describe("Article List", () => {
  it("should mount", () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(2);
  });
});
