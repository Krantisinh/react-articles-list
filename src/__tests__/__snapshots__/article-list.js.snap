// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List should mount 1`] = `
<div>
  <div
    style={
      Object {
        "border": "1px solid #ccc",
        "display": "flex",
        "flexDirection": "column",
        "fontFamily": "Verdana, Arial, Helvetica, sans-serif",
        "marginBottom": "1.5rem",
        "padding": "1rem",
      }
    }
  >
    <h2 />
    <div
      style={
        Object {
          "color": "gray",
          "fontSize": "0.75rem",
          "marginBottom": "0.5rem",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "marginBottom": "1rem",
        }
      }
    />
    <div
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "end",
        }
      }
    >
      -
       
      <a>
        undefined undefined
      </a>
    </div>
  </div>
  <div
    style={
      Object {
        "border": "1px solid #ccc",
        "display": "flex",
        "flexDirection": "column",
        "fontFamily": "Verdana, Arial, Helvetica, sans-serif",
        "marginBottom": "1.5rem",
        "padding": "1rem",
      }
    }
  >
    <h2 />
    <div
      style={
        Object {
          "color": "gray",
          "fontSize": "0.75rem",
          "marginBottom": "0.5rem",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "marginBottom": "1rem",
        }
      }
    />
    <div
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "end",
        }
      }
    >
      -
       
      <a>
        undefined undefined
      </a>
    </div>
  </div>
</div>
`;
